pkg/apis/core/validation/validation.go

```diff
@@ -3321,12 +3321,13 @@ func validateInitContainerRestartPolicy(restartPolicy *core.ContainerRestartPoli
 	if opts.AllowContainerRestartPolicyRules {
```

Comment:
If `restartPolicy == nil` this is all skipped - shouldn't it require that `len(restartRules) == 0` in that case?